src/app/_components/ClickTooltip.tsx

```diff
@@ -0,0 +1,64 @@
+'use client'
+
+import { useId, useState, type ReactNode } from 'react'
+
+import * as RadixPopover from '@radix-ui/react-popover'
+import { clsx } from 'clsx'
+
+type TooltipChildrenRenderProps = {
+  open: boolean
+  tooltipId: string
+}
+
+export type ClickTooltipProps = {
+  children: ReactNode | ((props: TooltipChildrenRenderProps) => ReactNode)
+  description: string
+  gap?: RadixPopover.PopoverContentProps['sideOffset']
+  side?: RadixPopover.PopoverContentProps['side']
+}
+
+const animationClasses = {
+  bottom: 'data-[state=open]:data-[side=bottom]:animate-slideUpAndFade',
+  left: 'data-[state=open]:data-[side=left]:animate-slideRightAndFade',
+  right: 'data-[state=open]:data-[side=right]:animate-slideLeftAndFade',
+  top: 'data-[state=open]:data-[side=top]:animate-slideDownAndFade',
+}
+
+export function ClickTooltip({
+  children,
+  description,
+  gap = 8,
+  side = 'top',
+}: ClickTooltipProps) {
+  const [open, setOpen] = useState(false)
+  const id = useId()
+
+  const tooltipId = `tooltip-${id}`
+
+  return (
+    <RadixPopover.Root open={open} onOpenChange={setOpen}>
+      <RadixPopover.Trigger asChild>
+        {typeof children === 'function'
+          ? children({ open, tooltipId })
+          : children}
+      </RadixPopover.Trigger>
+
+      <RadixPopover.Portal>
+        <RadixPopover.Content
+          hideWhenDetached
+          id={tooltipId}
+          sideOffset={gap}
+          side={side}
+          role="tooltip"
+          className={clsx(
+            'tooltip-content focus-visible:outline-none',
```

Comment:
Why is `tooltip-content` abstracted - and why is `focus-visible:outline-none` not?

Reply:
Ah yes, this should be part of `tooltip-content`!

Reply:
✅